src/to-string.cpp

```diff
@@ -3,9 +3,17 @@
 
 #include "types.h"
 
-#define STRCASE(T, X) case RS2_##T##_##X: {\
-        static const std::string s##T##_##X##_str = make_less_screamy(#X);\
-        return s##T##_##X##_str.c_str(); }
+#define STRX( X ) make_less_screamy( #X )
+#define STRCASE( T, X )                                                                                                \
+    case RS2_##T##_##X: {                                                                                              \
+        static const std::string s##T##_##X##_str = STRX( X );                                                         \
+        return s##T##_##X##_str.c_str();                                                                               \
+    }
+#define STRARR( T, X ) arr[RS2_##T##_##X] = STRX( X )
```

Comment:
This should be next to it's usage.
No reason to place a global macro that count on a specific object inside a function

Reply:
It's where the previous macro was, and can be used in any of the `to_string` functions. I just implemented it for one, but the context for it is file-wide.

Reply:
Oh, OK.
Well if `arr` is the problem, I made it:
`#define STRARR( ARRAY, T, X ) ARRAY[RS2_##T##_##X] = STRX( X )`

It's only for use in this file, and the only intended usage is supposed readability (same for the previous macro). You're right, most people will copy-paste. But I image we'll incrementally change all the functions to return strings and not `const char *`.

Reply:
In my next CR I'll update all the to_string functions and clean things up, if you agree -- but this will mean merge issues elsewhere, almost for sure.

Reply:
What is the motivation of this change?
Performance?
Notice that this change may affect our tools and wrappers.
Viewer use it.
And I believe wrapper do too (python for sure)

Reply:
Nobody uses these APIs -- they're internal to librealsense. It's the `rs2_<enum>_to_string` functions that are the APIs and use these internally, I.e., the change I made in this PR breaks no API.

Yes, it's performance. But you voiced an objection that makes sense, so if you don't disagree I'll clean things up to make everything 1. the same and 2. possibly easier to understand, though the mechanism (arrays initialized by a lambda, and still using macros) will still be a little not straightforward.